src/node/construct.rs

```diff
@@ -48,6 +49,42 @@ impl<J: Jet> ConstructNode<J> {
     pub fn arrow(&self) -> &Arrow {
         &self.data.arrow
     }
+
+    /// Sets the source and target type of the node to unit
+    pub fn type_as_program(&self) -> Result<(), types::Error> {
+        let unit_ty = types::Type::unit();
+        self.arrow()
+            .source
+            .unify(&unit_ty, "setting root source to unit")?;
+        self.arrow()
+            .target
+            .unify(&unit_ty, "setting root source to unit")?;
+        Ok(())
+    }
+
+    /// Convert a [`ConstructNode`] to a [`CommitNode`] by finalizing all of the types.
+    ///
+    /// Also sets the source and target type of this node to unit. This is almost
+    /// certainly what you want, since the resulting `CommitNode` cannot be further
+    /// composed, and needs to be 1->1 to go on-chain. But if you don't, call
+    /// [`Self::finalize_types_without_fixing`] instead.
+    pub fn finalize_types(&self) -> Result<Arc<CommitNode<J>>, types::Error> {
+        self.type_as_program()?;
+        self.finalize_types_without_fixing()
+    }
+
+    /// Convert a [`ConstructNode`] to a [`CommitNode`] by finalizing all of the types.
+    ///
+    /// Also sets the source and target type of this node to unit.
+    pub fn finalize_types_without_fixing(&self) -> Result<Arc<CommitNode<J>>, types::Error> {
```

Comment:
b3bce4e36dc504598954839cb0c9ab3556a2725e: Wrong documentation. This method does not enforce 1 → 1 expressions.

Also, we might want to name this method in accordance with `type_as_program` from above, to show the contrast. Maybe `finalize_types_no_program_type`.